### ISSUE TITLE
patch-package fails with git config user.useconfigonly

### ISSUE THREAD

**Issue by danilobuerger** [ds300/patch-package#76]
When using `user.useconfigonly` in git:
```
git config --global user.useconfigonly true
git config --global --unset-all user.email
```
`patch-package` can no longer create patches and fails with:
```
❯ yarn run patch-package some-package
yarn run v1.7.0
$ node_modules/.bin/patch-package some-package
☑ Creating temporary folder
☑ Building clean node_modules with yarn
☑ Diffing your files with clean files

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: no email was given and auto-detection is disabled
```

**Comment by ds300** [comment]
Hi! :wave: Thanks for the report! That makes sense, I wasn't aware of the `userconfigonly` option! I think it would fix this if patch-package adds a  `.git/config` file before creating the patch. I'll try that out.

**Comment by ds300** [comment]
this should have been fixed in #104
